tools/webassembly.py

```diff
@@ -237,7 +237,7 @@ def read_init(self):
       opcode = OpCode(self.read_byte())
       args = []
       if opcode in (OpCode.GLOBAL_GET, OpCode.I32_CONST, OpCode.I64_CONST):
-        args.append(self.read_uleb())
+        args.append(self.read_sleb())
```

Comment:
Why does this need to change?

Reply:
(it seems odd that a `global.get`index could be negative)

Reply:
The number that follows I32_CONST or I64_CONST instructions is stored as uninterpreted integers (https://webassembly.github.io/spec/core/syntax/values.html#syntax-int) which are signed LEBs in the binary format, and then reinterpreted as unsigned as needed.

Reply:
Those make sense, but this also affects the global.get. Perhaps we can separate the cases?

Reply:
Sounds reasonable.   Done.

(BTW I think maybe the distinction doesn't matter until we get over 2^31.. perhaps due to the way signed vs unsigned LEBs are encoded).